mlflow/spark/__init__.py

```diff
@@ -288,125 +286,29 @@ def log_model(
-            metadata=metadata,
-            params=params,
-            tags=tags,
-            model_type=model_type,
-            step=step,
-            model_id=model_id,
-        )
-
-    if not isinstance(spark_model, PipelineModel):
+    if not (_is_spark_connect_model(spark_model) or isinstance(spark_model, PipelineModel)):
         spark_model = PipelineModel([spark_model])
-    run_id = mlflow.tracking.fluent._get_or_start_run().info.run_id
-    run_root_artifact_uri = mlflow.get_artifact_uri()
-    remote_model_path = None
-    if _should_use_mlflowdbfs(run_root_artifact_uri):
-        remote_model_path = append_to_uri_path(run_root_artifact_uri, name, _SPARK_MODEL_PATH_SUB)
-        mlflowdbfs_path = _mlflowdbfs_path(run_id, name)
-        with databricks_utils.MlflowCredentialContext(
-            get_databricks_profile_uri_from_artifact_uri(run_root_artifact_uri)
-        ):
-            try:
-                spark_model.save(mlflowdbfs_path)
-            except Exception as e:
-                raise MlflowException("failed to save spark model via mlflowdbfs") from e
-
-    # If the artifact URI is a local filesystem path, defer to Model.log() to persist the model,
-    # since Spark may not be able to write directly to the driver's filesystem. For example,
-    # writing to `file:/uri` will write to the local filesystem from each executor, which will
-    # be incorrect on multi-node clusters.
-    # If the artifact URI is not a local filesystem path we attempt to write directly to the
-    # artifact repo via Spark. If this fails, we defer to Model.log().
-    elif (
-        is_local_uri(run_root_artifact_uri)
-        or databricks_utils.is_in_databricks_serverless_runtime()
-        or databricks_utils.is_in_databricks_shared_cluster_runtime()
-        or not _maybe_save_model(
-            spark_model,
-            append_to_uri_path(run_root_artifact_uri, name),
-        )
-    ):
-        dfs_tmpdir = dfs_tmpdir or MLFLOW_DFS_TMP.get()
-        _check_databricks_uc_volume_tmpdir_availability(dfs_tmpdir)
-        return Model.log(
-            artifact_path=artifact_path,
-            name=name,
-            flavor=mlflow.spark,
-            spark_model=spark_model,
-            conda_env=conda_env,
-            code_paths=code_paths,
-            dfs_tmpdir=dfs_tmpdir,
-            registered_model_name=registered_model_name,
-            signature=signature,
-            input_example=input_example,
-            await_registration_for=await_registration_for,
-            pip_requirements=pip_requirements,
-            extra_pip_requirements=extra_pip_requirements,
-            metadata=metadata,
-            params=params,
-            tags=tags,
-            model_type=model_type,
-            step=step,
-            model_id=model_id,
-        )
-    # Otherwise, override the default model log behavior and save model directly to artifact repo
-    logged_model = mlflow.initialize_logged_model(
+
+    return Model.log(
```

Comment:
This is the main change.

Reply:
> does this work for non-connect models

`save_model` works for them.

Reply:
Is there a special where a model can't be serialized in the local file system?

Reply:
for SparkML model, the API only supports `model.save(dbfs_path)` , so the `save_model` has to save it to a temp DBFS path (or UC volume path), then download it to local fs path.